did_peer/src/peer_did/peer_did/generic.rs

```diff
@@ -0,0 +1,144 @@
+use crate::{
+    error::DidPeerError,
+    peer_did::{
+        numalgos::{numalgo2::Numalgo2, numalgo3::Numalgo3, NumalgoKind},
+        parse::parse_numalgo,
+        validate::validate,
+    },
+};
+use did_parser::Did;
+use serde::{Deserialize, Deserializer, Serialize, Serializer};
+
+use super::PeerDid;
+
+#[derive(Clone, Debug, PartialEq)]
+pub enum GenericPeerDid {
+    Numalgo2(PeerDid<Numalgo2>),
+    Numalgo3(PeerDid<Numalgo3>),
+}
+
+impl GenericPeerDid {
```

Comment:
I would hope that (the current) `PeerDid` will be used as a go to and `GenericPeerDid` will be used when `PeerDid<N>` can't be used for some reason, so that's why I gave them shorter and longer name respectively.

Reply:
I see, alright then